### ISSUE TITLE
`Access denied for user 'nextcloud'@'xxx.xxx.xx.x' (using password: YES)` on a fresh install

### ISSUE THREAD

**Issue by F1zzyD** [nextcloud/docker#2235]
As the title states, I used the docker.yml example included on the set up page for docker containers and Nextcloud refuses to let me create an admin account. I have quadruple checked that the passwords match exactly and that I entered the information as detailed in the guide. No matter what permissions I change or directories I add Nextcloud will still say "Access denied", even on a fresh install.

Why is this happening?

Note: I left the password information blank for obvious reasons. Everything else comes from the official guide that says it will work....

[COMPOSE.TXT](https://github.com/user-attachments/files/15816950/COMPOSE.TXT)

**Comment by F1zzyD** [comment]
> The "Access denied" is presumably coming from MariaDB not Nextcloud.

I've gone through and deleted all Maria databases and I'm still receiving the error on a fresh install. Seems to be a problem with Nextcloud 

> Any chance you started up the stack (or at least the db container) once before? Such as with a different config/password.

I did, but as I said I've even done `docker system prune -a` when I shutdown Nextcloud. I'm putting in the fields exactly what's in the docker compose Nextcloud still gives me this:

> Error while trying to create admin account: An exception occurred in the driver: SQLSTATE[HY000] [1045] Access denied for user 'nextcloud'@'xxx.xxx.xx.x' (using password: YES) 

The logs say this: 

> [Warning] Access denied for user 'nextcloud'@'xxx.xxx.xx.x' (using password: YES)

Not sure how this is such a problem...

**Comment by F1zzyD** [comment]
Update: I changed the volumes from `db:/var/lib/mysql` and `  `nextcloud:/var/www/html` and it went and installed, but not to those directories. What would cause Nextcloud to take the volume changes but not actually use the directories I pointed it to?

**Comment by arialdomartini** [comment]
Getting exactly the same error.

**Comment by arialdomartini** [comment]
Works using the MySQL image instead of MariaDB.

**Comment by joshtrichards** [comment]
> I've gone through and deleted all Maria databases and I'm still receiving the error on a fresh install. Seems to be a problem with Nextcloud

Delete the databases is not enough. You have to delete the named volume `db` in Docker.

> I did, but as I said I've even done docker system prune -a when I shutdown Nextcloud. 

That command doesn't touch volumes.

> Update: I changed the volumes from `db:/var/lib/mysql` and `nextcloud:/var/www/html` and it went and installed, but not to those directories. What would cause Nextcloud to take the volume changes but not actually use the directories I pointed it to? There was nothing in those original directories to begin with.

I'm not sure what you mean by "those directories". Do you means the ones to right of the colon? Those aren't directories on your host. They're directories in the containers. You can't change them. 

You can, however, rename the named volumes (the left of the colon). That would have the same effect as deleting the named volumes created during a prior attempt to start things up... If that worked, then the cause was exactly as I suspected: you previously started up a mariadb container and it was re-using the prior db volume it had been associated with.

> Getting exactly the same error.
> Edit: I managed to make it work adding

@arialdomartini Well, yeah, that parameter requires a value as stated in the docs. It can't be left blank.

> Edit2: amazing, I cannot make it work anymore.

It's a one-time parameter. Please go familiar yourself with either the MariaDB or MySQL Docker image documentation.

> Works using the MySQL image instead of MariaDB.

Yes, because it's a fresh deployment of a new container again.

> Works using the MySQL image instead of MariaDB.

I'd encourage you both check out the Docker docs on volume management/etc. - e.g. 
- https://docs.docker.com/storage/

as well as the MariaDB and MySQL Docker image docs.

**Comment by F1zzyD** [comment]
I got it fixed by changing it to where I wanted to install, which for me was `home/user/nextcloud` where I store all of my docker containers.

After some fiddling around I got it up and running, but it would hang on loading a simple .doc file. I'm talking like 20 minutes of nothing. In the meantime, I was attempting to fix the "https" error so I could log in on my phone, but now nextcloud says I have no files on the specified mount point except for the computer it's hosting on. On my other computer it says "upload something!" but on the host server it shows the file. I can't even log out without Nextcloud throwing me an error.

I give up. This whole program is a mess and not ready for deployment. Gives me an SSL error when I'm accessing from a secure domain that the official guides say. Can't load a doc even with the office suite installed. What a mess.
